src/DotVVM.Framework/Utils/ReflectionUtils.cs

```diff
@@ -106,6 +106,29 @@ public static string ExtractMemberStringValue(object? item, string propertyName)
             return item + "";
         }
 
+        /// <summary>
+        /// Checks whether given instantiated type is compatible with the open generic type
+        /// </summary>
+        public static bool IsAssignableToGenericType(this Type givenType, Type genericType)
```

Comment:
It would be better if this method returned the mapping of concrete types to the generic arguments from `genericType`